packages/twenty-server/src/engine/core-modules/workspace/workspace.resolver.ts

```diff
@@ -223,11 +223,11 @@ export class WorkspaceResolver {
   @Query(() => CustomDomainDetails, { nullable: true })
   @UseGuards(WorkspaceAuthGuard)
   async getCustomDomainDetails(
-    @AuthWorkspace() { customDomain }: Workspace,
+    @AuthWorkspace() workspace: Workspace,
   ): Promise<CustomDomainDetails | undefined> {
-    if (!customDomain) return undefined;
-
-    return await this.domainManagerService.getCustomDomainDetails(customDomain);
+    return this.workspaceService.getCustomDomainDetailsAndToggleIsCustomDomainEnabled(
```

Comment:
I like the function naming that is saying it's doing two things.
However i'm a bit confused by the product behavior of the resolver: getCustomDomainDetails is actually doing an update, this looks wrong. Should we have to endpoints instead?

Reply:
and once we have two resolvers, do we still need getCustomDomainDetailsAndToggleIsCustomDomainEnabled ?

Reply:
It's an expected side effect that I could hide by removing the toggle from the name of the function.